pkg/pgmodel/new_migrate.go

```diff
@@ -0,0 +1,161 @@
+	return nil
+}
+
+func installExtensionAllBalls(db *pgx.Conn) error {
+	const stmt = "CREATE EXTENSION promscale SCHEMA _prom_ext VERSION '0.0.0'"
+	_, err := db.Exec(
+		context.Background(),
+		stmt,
+	)
+	return err
+}
+
+func Migrate(conn *pgx.Conn, appVersion VersionInfo, leaseLock *util.PgAdvisoryLock, extOptions extension.ExtensionMigrateOptions) error {
+	// At startup migrators attempt to grab the schema-version lock. If this
+	// fails that means some other connector is running. All is not lost: some
+	// other connector may have migrated the DB to the correct version. We warn,
+	// then start the connector as normal. If we are on the wrong version, the
+	// normal version-check code will prevent us from running.
+
+	if leaseLock != nil {
+		locked, err := leaseLock.GetAdvisoryLock()
+		if err != nil {
+			return fmt.Errorf("error while acquiring migration lock %w", err)
+		}
+		if !locked {
+			return MigrationLockError
+		}
+		defer func() {
+			_, err := leaseLock.Unlock()
+			if err != nil {
+				log.Error("msg", "error while releasing migration lock", "err", err)
+			}
+		}()
+	} else {
+		log.Warn("msg", "skipping migration lock")
+	}
+
+	migrateMutex.Lock()
+	defer migrateMutex.Unlock()
+
+	// if the old prom_schema_migrations table exists, then we need to apply any outstanding
+	// migrations from the old way of doing migrations, then transition to the new way
+	// the prom_schema_migrations table will be dropped as a part of the transition
+	schemaMigrationTableExists, err := doesSchemaMigrationTableExist(conn)
+	if err != nil {
+		return err
+	}
+	if schemaMigrationTableExists {
+		if err = oldMigrate(conn, appVersion); err != nil {
+			return fmt.Errorf("error while trying to migrate DB: %w", err)
+		}
+		if err = removeOldExtensionIfExists(conn); err != nil {
+			return fmt.Errorf("error while dropping old promscale extension: %w", err)
+		}
+		if err = installExtensionAllBalls(conn); err != nil {
+			return fmt.Errorf("error while installing promscale extension version 0.0.0: %w", err)
+		}
+	}
+
+	_, err = extension.InstallUpgradePromscaleExtensions(conn, extOptions)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func oldMigrate(db *pgx.Conn, versionInfo VersionInfo) (err error) {
+	appVersion, err := semver.Make(versionInfo.Version)
+	if err != nil {
+		return errors.ErrInvalidSemverFormat
+	}
+
+	mig := NewMigrator(db, migrations.MigrationFiles, tableOfContents)
+
+	err = mig.Migrate(appVersion)
+	if err != nil {
+		return fmt.Errorf("Error encountered during migration: %w", err)
+	}
+
+	return nil
+}
+
+// CheckDependencies makes sure the Promscale and TimescaleDB extensions are set up correctly. This will set
+// the ExtensionIsInstalled flag and thus should only be called once, at initialization.
+func CheckDependencies(db *pgx.Conn, migrationFailedDueToLockError bool, extOptions extension.ExtensionMigrateOptions) (err error) {
+	return extension.CheckVersions(db, migrationFailedDueToLockError, extOptions)
```

Comment:
do we want to just replace the CheckDependencies calls with extension.CheckVersions

Reply:
Yes.

Reply:
addressed in https://github.com/timescale/promscale/pull/1132/commits/62208e8414db47c633b49d0d48493eb50fd6a533